### ISSUE TITLE
Unnecessary variable enforcement on font-face declaration

### ISSUE THREAD

**Issue by Okazari** [AndyOGo/stylelint-declaration-strict-value#201]
Hello there ! I've recently run into an issue with the linter rule :smile: 

```css
@font-face {
  font-family: "Inter";
  src: url("assets/Inte.var.woff2") format("woff2");
  font-display: swap;
  font-weight: 100 900;
}
```

:point_up: In that case the linter wants the `font-weight` property to be a variable. I'm not sure that it makes sense as the `font-weight` range is tied with the font file we are using and will not change or be re-used.

WDYT ?

**Comment by AndyOGo** [comment]
@Okazari thanks for your message.

please provide more details, like what is your config, are you usibg css, sass or less syntax?

Ideally share a reproduction repo.

**Comment by Okazari** [comment]
We are using `sass` with this `stylelint.rc.js` config :
```js
module.exports = {
  "plugins": [
    "stylelint-order",
    "stylelint-scss",
    "stylelint-declaration-strict-value"
  ],
  "extends": "stylelint-config-recommended",
  "rules": {
   // A lot of rules tweaks
  }
}
```

I solve that by using a `sass variable` instead of `100 900` but I feel that it's a case were we doesn't need to enforce the use of a variable.

**Comment by AndyOGo** [comment]
Thank you for sharing your config.

You are using the default config.
Please make sure to read the documentation next time, before you open an incorrect issue report. This is intended for bugs, not questions or support.

**Comment by Okazari** [comment]
Sorry, I may have expressed myself poorly.
I wasn't looking for an answer as I already had make a linter compliant code
I was suggesting an improvement.

What I consider an issue, is that the plugin enforce us to use a variable somewhere where it doesn't make sense.
The `100 900` range value is tied to the `Inter` variable itself. We will not have to change it without changing the font.

Do you think we could make the linter allow row values for the `font-weight` (and some others) properties in `@font-face` declaration ?

**Comment by AndyOGo** [comment]
I understood your suggestion at the first time.

Please read the documentation.
You need to tell this plugin what and how to lint. If you dont do this then the default config applies.

btw. you can disable variables for soecific props, but this is up to you.
https://github.com/AndyOGo/stylelint-declaration-strict-value#disable-variables-for-specific-props

**Comment by silverwind** [comment]
This should still be fixed imho, `@font-face` not accepting `var()` is a well-known browser limitation and a real footgun. The plugin could look up the AST for any parent `@font-face` rules and not raise a error.

**Comment by AndyOGo** [comment]
Thank you.

The purpose of this plugin is to support multiple paradigms (CSS, SASS/SCSS, Less, etc.) and at the same time to not go into their respective contextual details (compiled, not compiled, browser support, etc.).

Inversion of control is applied, so that every user can configure this rule to match the needs and preferences of their project.

**Comment by silverwind** [comment]
Ok, but there is currently no way via config to say "don't flag when the declaration is inside a `@font-face` block".

**Comment by AndyOGo** [comment]
You are right, thank you for the hint.

**Comment by AndyOGo** [comment]
I see two options to approach your problem.

1. Using stylelints methods to [ignore code](https://stylelint.io/user-guide/ignore-code/) - IMHO preferred if this problem is a minority
2. Extend this rule to ignore rules that match within a certain AST tree, like an `@font-face` at-rule.
   - But how often does it happen?
   - How many users needs this?
   - And can't it be solved by [ignoring code](https://stylelint.io/user-guide/ignore-code/)?

**Comment by silverwind** [comment]
The root issue is that `var()` is invalid inside non-element at-rules, at least as far as major browsers are concerned.

Af few other such non-element at-rules are are `@import`, `@charset`, `@namespace`, `@font-face`, `@counter-style`, `@page`.

`@font-face` is somewhat commonly used to define custom fonts and its syntax looks exactly like a regular CSS rule but is technically a "configuration block" and this plugin seems to confuse the two.

I only encountered it once, currently using inline disable, but imho it points to a deeper problem inside this plugin's parser.

**Comment by AndyOGo** [comment]
Thank you.

You encounter this issue only when using vanilla CSS, right?
What is expected if PostCSS is used, to transform any of those `var()` inside a `@font-face` rule?

At the moment this plugin doesn't do any contextual parsing on purpose, it solely relies on the AST given by stylelint and walks all CSS properties that match the config, then checks for CSS/SCSS/Less/etc. syntaxes.

Anyway it's possible to skip a prop by checking it's context either by a selector or an at-rule.

**Comment by silverwind** [comment]
Yes, vanilla CSS, minimal reproduction:

```css
@font-face {
  font-weight: 500;
}
```

Rule config:

```json
"scale-unlimited/declaration-strict-value": [["font-weight"]]
```

If the stylelint ast gives indication whether you are inside `@font-face` or a "configuration block", it should be easy to fix. Can't answer PostCSS but stylelint configured very lightly with no custom syntax.

**Comment by AndyOGo** [comment]
@Okazari  Thank you very much for the original issue.
@silverwind  Thank you very much for upvoting and helping to justify adding it.

It's released with [v1.12.0](https://github.com/AndyOGo/stylelint-declaration-strict-value/releases/tag/v1.12.0) holding these [changes](https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/CHANGELOG.md#1120-2026-08-24).

**Comment by silverwind** [comment]
Thanks, I noticed one defect: `["@font-face"]` (which would be my go-to form) does not work. I'll go with regex form for now.

**Comment by AndyOGo** [comment]
Oh, I will add another test for arrays similar too
https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/test/at-rule-string.js

**Comment by silverwind** [comment]
I've gone with `{"@font-face": "font-weight"}` for now.

One thing that surprise me is that `/^@font-face$/` was not enough to ignore all props inside a `@font-face` because the plugin still required `font-weight` to be listed in `values`, but that seems to be unsolvable with the current design of the plugin.

**Comment by AndyOGo** [comment]
Released a fix for the array config [v1.12.1](https://github.com/AndyOGo/stylelint-declaration-strict-value/releases/tag/v1.12.1) holding these [changes](https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/CHANGELOG.md#1121-2026-08-24).

---

@silverwind To ignore all properties you could use the boolean hash config.
Example in this test: https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/test/at-rule-hash-boolean.js

However by default a simple string or list ignores all properties that are listed in the primary options.
Example in these tests:
- https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/test/at-rule-string.js
- https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/test/at-rule-regex.js
- https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/test/at-rule-list.js

**Comment by silverwind** [comment]
Thanks, I went with `ignoreAtRules: ['@font-face']` and `font-weight` in value list now, does exactly what I need.
